src/include/type/type_id.h

```diff
@@ -32,6 +32,7 @@ enum class TypeId : uint8_t {
   VARBINARY,
   PARAMETER_OFFSET,
   VARIADIC,
+  HACK_PG_TYPE_VAR_ARRAY,  ///< pg_type requires a distinct type for var_array.
```

Comment:
Is it a hack, then?

Reply:
Good question. I am concerned that we do not have support for this anywhere else in the system, though -- I suppose/hope it will blow up as "UNKNOWN TYPE" if people try to use it elsewhere.

Reply:
renamed to VAR_ARRAY